### PR TITLE
Changed UDP reading algorithm to preserve datagram integrity.

### PR DESCRIPTION
**Pull request by grzkv**
Followup to https://github.com/bookingcom/nanotube/pull/79#discussion_r446879044

Tested with production traffic emulation.
Benchmarks are OK.

### CODE REVIEW

File: cmd/nanotube/listen.go [hunk 1]
```diff
@@ -124,25 +124,28 @@ loop:
...
-			lg.Error("error reading a line from UDP connection", zap.Error(err))
+		nRead, err := conn.Read(buf)
+		if err != nil {
+			lg.Error("error reading UDP datagram", zap.Error(err))
 			continue
 		}
 
+		lines := bytes.Split(buf[:nRead], []byte{'\n'})
+
 		// TODO (grzkv): string -> []bytes
```
Comment (avereha) [inline cmd/nanotube/listen.go:145]: ` 		// TODO (grzkv): string -> []bytes  and pay attention to race conditions on "buf"`

File: cmd/nanotube/listen.go [hunk 2]
```diff
@@ -124,25 +124,28 @@ loop:
...
-		sendToMainQ(string(line), queue, ms)
+		for i := 0; i < len(lines)-1; i++ {
+			sendToMainQ(string(lines[i]), queue, ms)
```
Comment (avereha) [inline cmd/nanotube/listen.go:147]: When we get to implement string -> []bytes we will have to be careful about race conditions on "buf".
Split() on line 143 returns collections of slices in the same array(line 135).
If we send lines[] to another goroutine, we will have this goroutine updating buf(line 137) while other goroutine is reading it.
Added suggestion for comment so we remember about this in the future.

File: cmd/nanotube/listen.go (hunk 1 above)
Comment (grzkv) [reply cmd/nanotube/listen.go:145]: ;) Yes, yes, the `Split` does not copy.

Comment (grzkv) [reply cmd/nanotube/listen.go]: Added.

File: cmd/nanotube/listen.go (hunk 2 above)
Comment (grzkv) [reply cmd/nanotube/listen.go:148]: I am aware. Thanks. Added comments.

**Pull request merged**